resources/setup-spin

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+# This script uses sh instead of bash to be compatible with as many distros as possible.
+
+# The script is located in the Rancher Desktop resources/ directory.
+resources_dir=$(dirname "$0")
+
+# We run setup-spin in the rancher-desktop distro to setup spin on the Win32 host.
+if [ "$WSL_DISTRO_NAME" = "rancher-desktop" ]; then
+  app_data_dir=$(/bin/wslpath "$(powershell.exe -Command "Write-Output \${Env:LOCALAPPDATA}")" | tr -d "\r")
+  system_root=$(/bin/wslpath "$(powershell.exe -Command "Write-Output \${Env:SystemRoot}")" | tr -d "\r")
+  spin="${resources_dir}/win32/bin/spin.exe"
+elif [ "$(uname)" = "Linux" ]; then
+  app_data_dir="${XDG_DATA_HOME:-$HOME/.local/share}"
+  spin="${resources_dir}/linux/bin/spin"
+else
+  app_data_dir="${HOME}/Library/Application Support"
+  spin="${resources_dir}/darwin/bin/spin"
+fi
+
+if [ ! -x "$spin" ]; then
+  echo "Cannot execute '${spin}' (or does not exist)"
+  exit 1
+fi
+
+spin_dir="${app_data_dir}/spin"
+
+# shellcheck disable=SC2012 # Using `ls` is fine
+if [ -d "${spin_dir}/templates" ] && [ "$(ls -1 "${spin_dir}/templates" | wc -l)" -gt 0 ]; then
+  echo "'${spin_dir}/templates' already exists and is not empty"
+  exit 0
+fi
+
+# shellcheck disable=SC2012 # Using `ls` is fine
+if [ -d "${spin_dir}/plugins" ] && [ "$(ls -1 "${spin_dir}/plugins" | wc -l)" -gt 0 ]; then
+  echo "'${spin_dir}/plugins' already exists and is not empty"
+  exit 0
+fi
+
+if [ "$WSL_DISTRO_NAME" = "rancher-desktop" ]; then
+  echo "Waiting for github.com to become resolvable"
+  for _ in $(seq 30); do
+    curl --head --silent http://github.com >/dev/null
+    rc=$?; test $rc -ne 0 && echo "curl exit status is $rc"
+    if [ $rc -ne 6 ]; then
+      break
+    fi
+    sleep 2
+  done
+fi
+
+# The reason for this complexity is to be able to run on systems without git.
+# We do need either curl or wget to be on the PATH though.
```

Comment:
If we want all this, we should either download it from JS or golang.

Reply:
Yeah, maybe, but I think good enough for now. Realistically you should have either `curl` or `wget` in your distro if you expect to do any Wasm app development inside. How else will you test your app?